providers/coco.go

```diff
@@ -0,0 +1,93 @@
+package providers
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+        "errors"
+        "github.com/oauth2-proxy/oauth2-proxy/pkg/apis/sessions"
+        "github.com/oauth2-proxy/oauth2-proxy/pkg/requests"
+)
+
+type COCOProvider struct {
+	*ProviderData
+}
+
+//<!--COCO Begin-->
+//COCO: NewCOCOProvider initiates a new COCOProvider
+//<!--COCO End-->
```

Comment:
- just describe each functions and should be enought, no need to write coco begin and coco end in this file as the entire file is written by us only

Reply:
handled